GiftModels/FinancialDetails.cs

```diff
@@ -32,7 +32,28 @@ public class FinancialDetails
 
         public virtual string FundType { get; set; }
 
-        public virtual string FisAccount { get; set; }
+        public virtual string Purpose { get; set; }
+
+        public virtual string FisAccount {
+            get
+            {
+                string retval = null;
+                if (FundType.Equals("CURRENT") && !(Purpose.Equals("LOAN") || Purpose.Equals("CAPITAL")))

```

Comment:
@taylorkj you should check to make sure fundType or Purpose is not null, or else it'll throw an exception.  Alternately you could use `string.Equals(FundType, "CURRENT")`